tar/multitape/multitape_write.c

```diff
@@ -843,16 +843,33 @@ writetape_checkpoint(TAPE_W * d)
 		goto err0;
 
 	/*
+	 * Back up statistics about the metadata and metaindex;
+	 * this will be restored after the chunk layer writes the
+	 * directory file.
+	 */
+	chunks_write_extrastats_copy(d->C, 0);
+
+	/*
 	 * Flush data through and write the metaindex and metadata;
 	 * checkpoints are partial archives, so mark it as such.
+	 * This also adds the metadata and metaindex to the
+	 * "extra" statistics; we need these when we write the
+	 * directory file, but we will restore the original
+	 * statistics later.
```

Comment:
... since the metadata and metaindex from this checkpoint will be discarded if/when another checkpoint is created or the archive is completed.